shared-bindings/_protomatter/Protomatter.c

```diff
@@ -0,0 +1,288 @@
+        { MP_QSTR_doublebuffer, MP_ARG_BOOL, { .u_bool = true } },
+        { MP_QSTR_framebuffer, MP_ARG_OBJ, { .u_obj = mp_const_none } },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    // Because interrupt handlers point directly at protomatter objects,
+    // it is NOT okay to move them to the long-lived pool later.  Allocate
+    // them there to begin with, since generally they'll be long-lived anyway.
+    protomatter_protomatter_obj_t *self = &allocate_display_bus_or_raise()->protomatter;
+    self->base.type = &protomatter_Protomatter_type;
+
+    uint8_t rgb_count, addr_count;
+    uint8_t rgb_pins[MP_ARRAY_SIZE(self->rgb_pins)];
+    uint8_t addr_pins[MP_ARRAY_SIZE(self->addr_pins)];
+    uint8_t clock_pin = validate_pin(args[ARG_clock_pin].u_obj);
+    uint8_t latch_pin = validate_pin(args[ARG_latch_pin].u_obj);
+    uint8_t oe_pin = validate_pin(args[ARG_oe_pin].u_obj);
+
+    validate_pins(MP_QSTR_rgb_pins, rgb_pins, MP_ARRAY_SIZE(self->rgb_pins), args[ARG_rgb_list].u_obj, &rgb_count);
+    validate_pins(MP_QSTR_addr_pins, addr_pins, MP_ARRAY_SIZE(self->addr_pins), args[ARG_addr_list].u_obj, &addr_count);
+
+    mp_obj_t framebuffer = args[ARG_framebuffer].u_obj;
+    if (framebuffer == mp_const_none) {
+        int width = args[ARG_width].u_int;
+        int bufsize = 2 * width * rgb_count / 3 * (1 << addr_count);
+        framebuffer = mp_obj_new_bytearray_of_zeros(bufsize);
+    }
+
+    common_hal_protomatter_protomatter_construct(self,
+        args[ARG_width].u_int,
+        args[ARG_bit_depth].u_int,
+        rgb_count, rgb_pins,
+        addr_count, addr_pins,
+        clock_pin, latch_pin, oe_pin,
+        args[ARG_doublebuffer].u_bool,
+        framebuffer, NULL);
+
+    claim_and_never_reset_pins(args[ARG_rgb_list].u_obj);
+    claim_and_never_reset_pins(args[ARG_addr_list].u_obj);
+    claim_and_never_reset_pin(args[ARG_clock_pin].u_obj);
+    claim_and_never_reset_pin(args[ARG_oe_pin].u_obj);
+    claim_and_never_reset_pin(args[ARG_latch_pin].u_obj);
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|   .. method:: deinit
+//|
+//|     Free the resources (pins, timers, etc.) associated with this
+//|     protomatter instance.  After deinitialization, no further operations
+//|     may be performed.
+//|
+STATIC mp_obj_t protomatter_protomatter_deinit(mp_obj_t self_in) {
+    protomatter_protomatter_obj_t *self = (protomatter_protomatter_obj_t*)self_in;
+    common_hal_protomatter_protomatter_deinit(self);
+    return mp_const_none;
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(protomatter_protomatter_deinit_obj, protomatter_protomatter_deinit);
+
+static void check_for_deinit(protomatter_protomatter_obj_t *self) {
+    if (!self->core.rgbPins) {
+        raise_deinited_error();
+    }
+}
+
+//|   .. attribute:: paused
+//|
+//|     When paused, the matrix is not driven and all its LEDs are unlit.
```

Comment:
I'm not sure this needs to be separated from brightness of the display. It is the same as `brightness = 0`.